packages/genesys-spark-components/src/components/beta/gux-popover-beta/gux-popover.tsx

```diff
@@ -0,0 +1,321 @@
+        findElementById(this.root, this.for),
+        this.popupElement,
+        () => this.updatePosition(),
+        {
+          ancestorScroll: true,
+          elementResize: true,
+          animationFrame: true,
+          ancestorResize: true
+        }
+      );
+    } else {
+      this.disconnectedCallback();
+    }
+  }
+
+  private updatePosition(): void {
+    const forElement = findElementById(this.root, this.for);
+
+    if (this.popupElement && forElement) {
+      void computePosition(forElement, this.popupElement, {
+        placement: this.position,
+        middleware: [
+          offset(7),
+          flip(),
+          shift(),
+          arrow({
+            element: this.arrowElement,
+            padding: 16
+          })
+        ]
+      }).then(({ x, y, middlewareData, placement }) => {
+        Object.assign(this.popupElement.style, {
+          left: `${x}px`,
+          top: `${y}px`
+        });
+
+        const side = placement.split('-')[0];
+
+        const staticSide = {
+          top: 'bottom',
+          right: 'left',
+          bottom: 'top',
+          left: 'right'
+        }[side];
+
+        const arrowRotation = {
+          top: 0,
+          right: 90,
+          bottom: 180,
+          left: -90
+        }[side];
+
+        // This is 13 because this makes the arrow look aligned when horizontal
+        // or 15 if vertical due to extra padding needed to show the shadow.
+        const arrowLen = side === 'left' || side === 'right' ? 15 : 13;
+
+        if (middlewareData.arrow) {
+          let x = middlewareData.arrow.x;
+          const y = middlewareData.arrow.y;
+
+          if (side === 'left' || side === 'right') {
+            x = x + 4;
+          }
+
+          this.popupElement.setAttribute('data-placement', placement);
+
+          Object.assign(this.arrowElement.style, {
+            left: x != null ? `${x}px` : '',
+            top: y != null ? `${y}px` : '',
+            right: '',
+            bottom: '',
+            [staticSide]: `${-arrowLen}px`,
+            transform: `rotate(${arrowRotation}deg)`
+          });
+        }
+      });
+    }
+  }
+
+  private dismiss(): void {
+    const dismissEvent = this.guxdismiss.emit();
+    if (!dismissEvent.defaultPrevented) {
+      this.isOpen = false;
+      this.popupElement.hidePopover();
+    }
+  }
+
+  private logForAttributeError(): void {
+    if (this.root.isConnected) {
+      console.error(
+        `gux-popover: invalid element supplied to 'for': "${this.for}"`
+      );
+    }
+  }
+
+  connectedCallback(): void {
+    this.updateForElement();
+    trackComponent(this.root, { variant: this.position });
+  }
+
+  componentDidLoad(): void {
+    if (!this.forElement) {
+      return;
+    }
+
+    this.popupElement.popover = 'manual';
+    this.forElement.popoverTargetElement = this.popupElement;
+    this.forElement.popoverTargetAction = 'toggle';
+
+    if (this.isOpen) {
+      this.popupElement.showPopover();
```

Comment:
From reading through the docs the `popupElement` should have an attribute of `popup` for the `showPopover` to work but I could not find it. 

https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/showPopover

Reply:
I think it's that the div that has a `popover` attribute on it should work to use the `showPopover` method.